src/application/service_tester.go

```diff
@@ -110,20 +103,14 @@ func (s *ServiceTester) getDockerComposeConfig() (types.DockerConfigs, error) {
 	return dockerConfigs, nil
 }
 
-func (s *ServiceTester) getServiceNames() []string {
-	return []string{s.DirName}
-}
-
-func (s *ServiceTester) getServiceOnlineTexts() map[string]string {
-	return map[string]string{
-		"": fmt.Sprintf("%s exited with code", s.DirName),
-	}
-}
-
 // Run runs the tests for the service and return true if the tests passed
 // and an error if any
 func (s *ServiceTester) Run() (int, error) {
-	err := composerunner.Run(s.RunOptions)
+	cmdPlus, err := composerunner.Run(s.RunOptions)
+	if err != nil {
+		return 1, err
+	}
+	err = cmdPlus.WaitForText(fmt.Sprintf("%s exited with code", s.DirName), time.Hour)
```

Comment:
How do we plan on handling this after switching to `exec`?

Reply:
Hmm, I would guess we'll need to poll docker checking if the container has exited?

Reply:
Actually docker-compose has a pretty useful option `--abort-on-container-exit`